Week-11/redux-2/src/components/common/Button.jsx

```diff
@@ -0,0 +1,58 @@
+import { memo } from "react";
+
+import styles from "./button.module.css";
+
+const marginMapping = {
+  mt: "marginTop",
+  mb: "marginBottom",
+  ml: "marginLeft",
+  mr: "marginRight",
+};
+const paddingMapping = {
+  pt: "paddingTop",
+  pb: "paddingBottom",
+  pl: "paddingLeft",
+  pr: "paddingRight",
+};
+
+const Button = ({
+  children,
+  bgColor,
+  textColor = "#000",
+  fontWeight = "normal",
+  ...props
+}) => {
+  console.log("rerender check");
+  const getPaddingOrMargin = () => {
+    const styles = {};
+    const allProperties = [
+      ...Object.keys(marginMapping),
+      ...Object.keys(paddingMapping),
+    ];
+    allProperties.forEach((property) => {
+      if (props[property]) {
+        const actualProperty =
+          marginMapping[property] || paddingMapping[property];
+        styles[actualProperty] = props[property];
+      }
+    });
+    return styles;
+  };
+  getPaddingOrMargin();
```

Comment:
If this function is going to be called during component mount, can we wrap it with an `useEffect ` ?